src/core/core.interaction.js

```diff
@@ -172,7 +173,7 @@ module.exports = {
 			var items = options.intersect ? getIntersectItems(chart, position) : getNearestItems(chart, position, false, distanceMetric);
 
 			if (items.length > 0) {
-				items = chart.getDatasetMeta(items[0]._datasetIndex).data;
+				items = [{datasetIndex: items[0].datasetIndex}]; // when mode: 'dataset' we only need to return datasetIndex
```

Comment:
Ok, missed this. Should probably be only visible items that are returned here, but I'm ok with this optimization for now.